src/ring_theory/integral_closure.lean

```diff
@@ -254,6 +254,20 @@ theorem mem_integral_closure_iff_mem_fg {r : A} :
 
 variables {R} {A}
 
+/-- Mapping an integral closure along an `alg_equiv` gives the integral closure. -/
+lemma integral_closure_map_alg_equiv (f : A ≃ₐ[R] B) :
+  (integral_closure R A).map (f : A →ₐ[R] B) = integral_closure R B :=
+begin
+  ext y,
+  rw subalgebra.mem_map,
+  split,
+  { rintros ⟨x, hx, rfl⟩,
+    exact is_integral_alg_hom f hx },
+  { intro hy,
+    use [f.symm y, is_integral_alg_hom (f.symm : B →ₐ[R] A) hy],
+    simp }
+end
```

Comment:
Should a machine be making those proofs?

Reply:
Sorry, I'm not sure what your question is. 

If you're asking whether the proof can be made more manual than it is: yes, the following proof term also works.
```lean
subalgebra.ext (λ y,
  ⟨λ ⟨x, hx, fx_eq⟩, fx_eq ▸ is_integral_alg_hom f hx,
   λ hy, ⟨f.symm y, is_integral_alg_hom (f.symm : B →ₐ[R] A) hy, f.apply_symm_apply y⟩⟩)
```

If you're asking whether the proof should be made less manual than it is right now: I don't have any idea for a tactic that would help here.

Reply:
@kbuzzard :arrow_up:

Reply:
I mean that you're proving that a predicate stays true along an isomorphism, and in my mind it's frustrating that this is not proved using automation. Is there some missing tactic? Still, this is not the problem of this PR.

Reply:
Perhaps a smarter `equiv_rw`/`transport` could do it, but we would have to teach it that when we have `f : alg_equiv`, it can manipulate subalgebras via `subalgebra.map f`, that this is coherent with the original `alg_equiv`, etc. So it would be automatable with lot of nontrivial work I'd expect.